server/src/bots/telegram/botResponse/trendResponse.ts

```diff
@@ -1,35 +1,76 @@
 import { getCovidTrends } from '../../../services/api/api-chart';
 import { addDays, Now } from '../../../utils/dateUtils';
 import { CountrySituationInfo } from '../../../models/covid19.models';
-import { Transform } from '../../../services/domain/chart';
+import {
+    Transform,
+    enrichWithTitle,
+    enrichWithType,
+} from '../../../services/domain/chart';
 import { catchAsyncError } from '../../../utils/catchError';
 import { getRequestedCountry } from '../../../services/domain/countries';
 import { logger } from '../../../utils/logger';
 import { CallBackQueryHandlerWithCommandArgument } from '../models';
 import * as TelegramBot from 'node-telegram-bot-api';
+import { Frequency } from './../../../models/constants';
 
 export const trendsByCountryResponse: CallBackQueryHandlerWithCommandArgument = async (
     bot: TelegramBot,
     message: TelegramBot.Message,
     chatId: number,
-    requestedCountry?: string | undefined
+    requestedCountry?: string | undefined,
+    requestedFrequency?: Frequency | undefined
 ): Promise<TelegramBot.Message> => {
+    requestedFrequency = requestedFrequency || Frequency.Weekly;
+
     const [err, [foundCountry, foundSituation]] = await catchAsyncError(
         getRequestedCountry(requestedCountry)
     );
     if (err) {
         logger.error(
             `Error ocured while trying to get requested country ${requestedCountry}`,
-            err);
+            err
+        );
 
         return bot.sendMessage(chatId, err.message);
     }
-    const lastWeekSituation = foundSituation.filter(
-        (c: CountrySituationInfo) => {
+
+    let startDate: Date;
+    let hasFilter = true;
+    switch (requestedFrequency) {
+        case Frequency.Weekly:
+            startDate = addDays(Now, -7);
+            break;
+        case Frequency.Monthly:
+            startDate = addDays(Now, -30);
+            break;
+        case Frequency.WholePeriod:
+            hasFilter = false;
+            break;
+    }
+
+    let periodSituation = foundSituation;
+    if (hasFilter) {
+        periodSituation = periodSituation.filter((c: CountrySituationInfo) => {
             const date = new Date(c.date);
-            return date < Now && date > addDays(Now, -7);
-        }
+            return date < Now && date > startDate;
+        });
+    }
+
+    const frequencyName =
+        requestedFrequency === Frequency.WholePeriod
+            ? 'Whole period'
+            : capitalize(requestedFrequency);
+
+    let model = enrichWithTitle(
```

Comment:
is that really a model? Maybe a chart? Does it have a type?

Reply:
###### Types
##### Types
#### Types
### Types
## Types
# Types
# kript
## kript
### kript
#### kript
##### kript
###### kript

Reply:
@danbilokha 
There is no need to add any type there. 
TypeScript is clever **enough** to identify and follow the type by return type of the function

Reply:
@danbilokha try to hover your mouse on the variable. And you'll see the macig.
I  know I promised to provide with a nice article on that point.
But so far I've just found.  [this](https://levelup.gitconnected.com/essential-typescript-types-get-started-with-typescript-5e20a5106c90)
![image](https://user-images.githubusercontent.com/25850290/79903908-8d222f80-841c-11ea-9d4b-9d8b644dfdfd.png)